atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/memory/InMemoryAsyncAtlasDbConfig.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.memory;
+
+import java.util.Optional;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.auto.service.AutoService;
+import com.palantir.atlasdb.spi.KeyValueServiceConfig;
+
+@JsonTypeName(InMemoryAsyncAtlasDbConfig.TYPE)
+@AutoService(KeyValueServiceConfig.class)
+public final class InMemoryAsyncAtlasDbConfig implements KeyValueServiceConfig {
```

Comment:
immutables?

Reply:
Just used InMemoryAtlasDbConfig as template. If anything, this should really be a singleton enum?